src/App.tsx

```diff
@@ -14,11 +17,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[]
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  chooseItemHandler = (item: string) => {
+    if (!(this.state.selectedGoods.includes(item))) {
+      this.setState(prevState => ({
+        selectedGoods: [...prevState.selectedGoods, item],
+      }));
+    }
+  };
+
+  RemoveItemHandler = (item: string) => {
+    this.setState(prevState => ({
+      selectedGoods: prevState.selectedGoods.filter(element => element !== item),
+    }));
+  };
+
+  resetSelectedItems() {
+    this.setState({
+      selectedGoods: [],
+    });
+  }
+
+  render() {
+    const { selectedGoods } = this.state;
+    let SelectedValues = '';
+
+    switch (selectedGoods.length) {
+      case 1:
+        SelectedValues = `${selectedGoods} is selected`;
+        break;
+
+      case 2:
+        SelectedValues = `${selectedGoods[0]} and ${selectedGoods[1]} are selected`;
+        break;
+
+      case 3:
+        SelectedValues = `${selectedGoods[0]},${selectedGoods[1]}  and ${selectedGoods[2]} are selected`;
+        break;
+      case 0:
+        SelectedValues = 'No items selected';
+        break;
+
+      default:
+        SelectedValues = `${selectedGoods} are selected`;
+        break;
+    }
```

Comment:
For this logic with formatting title, create a separate method e. g. `displaySelectedGoods()`. Method `render()` must be as clear as possible.